certbot-nginx/certbot_nginx/_internal/configurator.py

```diff
@@ -424,10 +436,11 @@ def _get_default_vhost(self, domain, allow_port_mismatch, port):
 
         # TODO: present a list of vhosts for user to choose from
 
-        raise errors.MisconfigurationError("Could not automatically find a matching server"
-            " block for %s. Set the `server_name` directive to use the Nginx installer." % domain)
+        raise errors.MisconfigurationError("Could not automatically find a matching server "
+                                           f"block for {domain}. Set the `server_name` directive "
+                                           "to use the Nginx installer.")
 
-    def _get_ranked_matches(self, target_name):
+    def _get_ranked_matches(self, target_name: str) -> List[Dict[str, Any]]:
```

Comment:
Going in that direction, the most elegant approach would be to go to a dataclass, that I find superior to embed a structured and typed set of elements. Dataclasses are a little cumbersome to define on Python <3.7, but as soon as we drop Python 3.6 we can take advantage of @dataclass decorator (https://docs.python.org/3/library/dataclasses.html). On Python 3.7, we can take advantage of `typing_extension` module that exposes `TypeDict` also.

Globally my preference would go to a dataclass here, what do you think ?

Reply:
@alexzorin since this is still theoritical as we have to support Python 3.6 and 3.7, I consider this discussion closed, but do not hesitate to comment otherwise.